python/pyspark/pandas/frame.py

```diff
@@ -2648,123 +2648,6 @@ def to_latex(
-            String, path object (implementing ``os.PathLike[str]``), or file-like
-            object implementing a binary ``write()`` function.
-        convert_dates : dict
-            Dictionary mapping columns containing datetime types to stata
-            internal format to use when writing the dates. Options are 'tc',
-            'td', 'tm', 'tw', 'th', 'tq', 'ty'. Column can be either an integer
-            or a name. Datetime columns that do not have a conversion type
-            specified will be converted to 'tc'. Raises NotImplementedError if
-            a datetime column has timezone information.
-        write_index : bool
-            Write the index to Stata dataset.
-        byteorder : str
-            Can be ">", "<", "little", or "big". default is `sys.byteorder`.
-        time_stamp : datetime
-            A datetime to use as file creation date.  Default is the current
-            time.
-        data_label : str, optional
-            A label for the data set.  Must be 80 characters or smaller.
-        variable_labels : dict
-            Dictionary containing columns as keys and variable labels as
-            values. Each label must be 80 characters or smaller.
-        version : {{114, 117, 118, 119, None}}, default 114
-            Version to use in the output dta file. Set to None to let pandas
-            decide between 118 or 119 formats depending on the number of
-            columns in the frame. Version 114 can be read by Stata 10 and
-            later. Version 117 can be read by Stata 13 or later. Version 118
-            is supported in Stata 14 and later. Version 119 is supported in
-            Stata 15 and later. Version 114 limits string variables to 244
-            characters or fewer while versions 117 and later allow strings
-            with lengths up to 2,000,000 characters. Versions 118 and 119
-            support Unicode characters, and version 119 supports more than
-            32,767 variables.
-        convert_strl : list, optional
-            List of column names to convert to string columns to Stata StrL
-            format. Only available if version is 117.  Storing strings in the
-            StrL format can produce smaller dta files if strings have more than
-            8 characters and values are repeated.
-        value_labels : dict of dicts
-            Dictionary containing columns as keys and dictionaries of column value
-            to labels as values. Labels for a single variable must be 32,000
-            characters or smaller.
-
-        Examples
-        --------
-        >>> df = ps.DataFrame({'animal': ['falcon', 'parrot', 'falcon', 'parrot'],
-        ...                    'speed': [350, 18, 361, 15]})
-        >>> df.to_stata('animals.dta')  # doctest: +SKIP
-        """
-        # Make sure locals() call is at the top of the function so we don't capture local variables.
-        args = locals()
-
-        return validate_arguments_and_invoke_function(
```

Comment:
Actually I think we should keep the implementation as is here, and just add an if-else with `compute.pandas_fallback` (and treat those as just special cases) ....

By using the fallback one, it will write the file into one of the executors, which wouldn't be quite useful ..

Reply:
got it, will update it tmr

Reply:
I still feel a bit hesitant about how to change now.
the newly added `to_hdf/to_feather/to_stata` methods are actually consistent with existing `to_excel/to_markdown/etc` that also load all the data to the driver.

I think there are 3 options:
1, let `to_hdf/to_feather/to_stata` alone;
2, hide those methods, no docs, users can not see those methods, only usable when set the config (this PR);
3, just add if-else with the config in the implementations;

wdyt? @ueshin @xinrong-meng @HyukjinKwon

Reply:
hmmmm ... okay .. I am fine with leaving it as is for now ....